kit/ChildSession.cpp

```diff
@@ -2506,13 +2509,20 @@ bool ChildSession::renderSlide(const StringVector& tokens)
     assert(bufferHeight <= suggestedHeight);
 
     bool done = false;
+    SlideCompressor scomp(_docManager->getSyncPool());
     while (!done)
     {
-        success = renderNextSlideLayer(bufferWidth, bufferHeight, devicePixelRatio, done);
+        success = renderNextSlideLayer(scomp, bufferWidth, bufferHeight, devicePixelRatio, done);
         if (!success)
             break;
     }
 
+    scomp.compress([this](const std::vector<char>& output) {
+        LOG_TRC("Sending response (" << output.size() << " bytes) for: " <<
+                std::string(output.data(), Util::findInVector(output, "\n") - 1));
```

Comment:
b: so do we have a problem here with a len arg of of npos-1 if that (a) encodeSubBufferToPNG failure happens?

Reply:
The compress method doesn't call us back in this case.

Reply:
ah yes, if (it->_output.size() > 0), fair enough